packages/transaction-controller/src/utils/swaps.ts

```diff
@@ -410,15 +428,19 @@ function updateSwapAndSendTransaction(
  * @param propsToUpdate - Properties to update
  * @param propsToUpdate.type - Type of the transaction
  * @param propsToUpdate.sourceTokenSymbol - Symbol of the token to be swapped
+ * @param isSwapAndSend - if this approval is for a swap and send
  * @returns The updated transaction meta object.
  */
 function updateSwapApprovalTransaction(
   transactionMeta: TransactionMeta,
   { type, sourceTokenSymbol }: Partial<TransactionMeta>,
+  isSwapAndSend = false,
 ): TransactionMeta {
   validateIfTransactionUnapproved(
     transactionMeta,
-    'updateSwapApprovalTransaction',
+    isSwapAndSend
```

Comment:
The purpose of this is just to throw a custom error to track where the update came from.

As this function doesn't actually exist, is there any need to overwrite the name at all?